detection_rules/rule_validators.py

```diff
@@ -40,31 +40,36 @@ def validate(self, data: QueryRuleData, meta: RuleMeta) -> None:
             packages_manifest = load_integrations_manifests()
             package_integrations = TOMLRuleContents.get_packaged_integrations(data, meta, packages_manifest)
 
+            # validate the query against fields within beats
+            self.validate_beats(data, meta)
+
             if package_integrations:
                 # validate the query against related integration fields
                 self.validate_integration(data, meta, package_integrations)
-            else:
-                for stack_version, mapping in meta.get_validation_stack_versions().items():
-                    beats_version = mapping['beats']
-                    ecs_version = mapping['ecs']
-                    err_trailer = f'stack: {stack_version}, beats: {beats_version}, ecs: {ecs_version}'
-
-                    beat_types, beat_schema, schema = self.get_beats_schema(data.index or [],
-                                                                            beats_version, ecs_version)
-
-                    try:
-                        kql.parse(self.query, schema=schema)
-                    except kql.KqlParseError as exc:
-                        message = exc.error_msg
-                        trailer = err_trailer
-                        if "Unknown field" in message and beat_types:
-                            trailer = f"\nTry adding event.module or event.dataset to specify beats module\n\n{trailer}"
-
-                        raise kql.KqlParseError(exc.error_msg, exc.line, exc.column, exc.source,
-                                                len(exc.caret.lstrip()), trailer=trailer) from None
-                    except Exception:
-                        print(err_trailer)
-                        raise
+
+    def validate_beats(self, data: QueryRuleData, meta: RuleMeta) -> None:
```

Comment:
```suggestion
    def validate_stack_combos(self, data: QueryRuleData, meta: RuleMeta) -> None:
```

or something similar to reflect the fact that it validates from the `stack_schema_map`